### PR TITLE
backend: Remove folders first in CLI

### PR DESCRIPTION
**Pull request by tianjing-li**
Thank you for contributing to the Cohere Toolkit!

- [ ] **PR title**: "area: description"
  - Where "area" is whichever of interface, frontend, model, tools, backend, etc. is being modified. Use "docs: ..." for purely docs changes, "infra: ..." for CI changes.
  - Example: "deployment: add Azure model option"

- [ ] **PR message**: ***Delete this entire checklist*** and replace with
    - **Description:** a description of the change
    - **Issue:** the issue # it fixes, if applicable
    - **Dependencies:** any dependencies required for this change

- [ ] **Add tests and docs**: Please include testing and documentation for your changes
- [ ] **Lint and test**: Run `make lint` and `make run-tests`

**AI Description**

The pull request introduces a new function, `delete_config_folders`, to handle an edge case where pre-existing configuration and secrets folders are present in the config folder. 

**Changes:**
- Adds the `delete_config_folders` function to `src/backend/cli/setters.py`.
- Calls the `delete_config_folders` function in `src/backend/cli/main.py` before writing template and configuration files.

### CODE REVIEW

**Review by malexw — APPROVED** [review]

File: src/backend/cli/main.py [hunk 1]
```diff
@@ -66,6 +67,9 @@ def start():
     write_env_file(secrets)
 
     # SET UP YAML CONFIG FILES
+    # Deal with strange edge case where there are preexisting configuration.yaml
+    # and secrets.yaml FOLDERS presents in the config folder - cause still unclear
+    delete_config_folders()
```
[the remark this answers is not in the document]
Comment (tianjing-li) [reply src/backend/cli/main.py:72]: That would make sense, I suppose we can have a follow up PR that checks on startup for these files as well
